qa_12 Sprint_4 final_project

MariaKassianenko opened the pull request:
Yandex_qa_12 
Sprint_4 
final_project 
pull_request

MariaKassianenko committed:
qa_12 Sprint_4 final_project

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,12 @@
+
+
+@pytest.fixture
+def user():
```
Можно лучше: это user, тут пользователя нет совсем. Лучше использовать для этой фикстуры имя driver или browser

AlexKitIt commented on tests/test_logo_links.py:
```diff
@@ -0,0 +1,25 @@
+from urls import yandex_page, main_page
+
+
+class TestButtons:
```
Можно лучше: над классом обычно размещают allure feature или suite

AlexKitIt commented on tests/test_logo_links.py:
```diff
@@ -0,0 +1,25 @@
+        order_page_user = OrderPage(user)
+        order_page_user.get_order_page()
+        order_page_user.click_scooter_logo()
+        assert order_page_user.url_current() == main_page
```
Можно лучше: если добавить assert message будет проще понять что именно и где у нас упало

AlexKitIt commented on variables.py:
```diff
@@ -0,0 +1,22 @@
+timeout_for_webdriverwait = 5
```
Можно лучше: это лучше хранить прямо в Base, тоже будет одно место для поддержки

AlexKitIt commented on pages/order_page.py:
```diff
@@ -0,0 +1,55 @@
+class OrderPage(BasePage):
+    @allure.step('Открыть страницу для оформления заказа')
+    def get_order_page(self):
+        self.user.get(order_page)
+        self.user.find_element(*OrderPageLoc.next_button)
```
Можно лучше: эти два метода (как и все прямые обращения к драйверу, в твоем случае к user) лучше вынести в base, так гораздо эстетичнее будет

AlexKitIt commented on pages/order_page.py:
```diff
@@ -0,0 +1,55 @@
+        self.user.get(order_page)
+        self.user.find_element(*OrderPageLoc.next_button)
+
+    @allure.step('Заполнить форму заказа')
```
Можно лучше: в отчете у тебя это шаг неделимый получился. Значит заполнение будет просто с указанием атрибутов. А если у тебя будет несколько отдельных stepов с заполнением имени, фамилии и т.д. в отчете это тоже будут отдельные шаги и если поменяется локатор одного из них - конкретно поймем что именно нужно править. Т.е. общий то метод верный, можно из self.user.find_element(*OrderPageLoc.name_field).send_keys(name) сделать метод "Ввести имя" и т.д.

MariaKassianenko committed:
qa_12 Sprint_4 final_project
a few fixes: more POM, less webdriver in page methods; assert messages added; allure.feature names added; order test got more detailed step - "fill order form", sub-steps added.
